### PR TITLE
Migrate last commands

### PR DESCRIPTION
**Pull request by SIMULATAN**
# Migrate last commands
### Checklist

- [X] I have checked the PRs for upcoming features/bug fixes.
- [X] I have read the [contributing guidelines](/CONTRIBUTING.md).
- [X] I agree with the [license](/LICENSE).

### Changes
- [X] Refactoring
- [X] Internal code
- [ ] External Code / Libraries
- [ ] Documentation
- [ ] Other: \_____

Closes Issue: #183 #186 #185

## Description
This is just a PR for me to fix small issues I found with #183

### CODE REVIEW

File: src/main/java/com/github/black0nion/blackonionbot/bot/BotInformation.java [hunk 1]
```diff
@@ -20,23 +20,29 @@
 import static com.github.black0nion.blackonionbot.misc.OperatingSystem.*;
 
 public class BotInformation {
+	private BotInformation() {}
 
-	private static final String datePattern = "dd.MM.yyyy HH:mm";
-	public static final SimpleDateFormat DATE_PATTERN = new SimpleDateFormat(datePattern);
-	public static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter.ofPattern(datePattern);
+	private static final String PATTERN = "dd.MM.yyyy HH:mm";
```
Comment (RealYusufIsmail) [inline src/main/java/com/github/black0nion/blackonionbot/bot/BotInformation.java:26]: I changed to osBean as sonarlight recommended and it is best practice.

**Review by RealYusufIsmail — CHANGES_REQUESTED** [review]

File: src/main/java/com/github/black0nion/blackonionbot/bot/BotInformation.java (hunk 1 above)
Comment (SIMULATAN) [reply src/main/java/com/github/black0nion/blackonionbot/bot/BotInformation.java:26]: osBean is a constant and should thus be called `OS_BEAN`, according to the java conventions

**Review by RealYusufIsmail — APPROVED** [review]

**Comment by RealYusufIsmail** [comment]
@SIMULATAN What is the purpose of this new reloadable annotation

File: src/main/java/com/github/black0nion/blackonionbot/api/impl/get/GetCommands.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+import com.github.black0nion.blackonionbot.bot.SlashCommandBase;
+import com.github.black0nion.blackonionbot.oauth.DiscordUser;
+import io.javalin.http.Context;
+import org.json.JSONObject;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+import java.util.Map;
+
+public class GetCommands implements IGetRoute {
```
Comment (RealYusufIsmail) [inline src/main/java/com/github/black0nion/blackonionbot/api/impl/get/GetCommands.java:14]: Add Java doc explaining what this.

**Review by RealYusufIsmail — CHANGES_REQUESTED** [review]

**Comment by SIMULATAN** [comment]
> @SIMULATAN What is the purpose of this new reloadable annotation

It makes it a method that can be reloaded via the `/reload` command

File: src/main/java/com/github/black0nion/blackonionbot/commands/fun/GiveawayCommand.java [hunk 3]
```diff
@@ -0,0 +1,129 @@
...
+					new OptionData(OptionType.INTEGER, DAYS, "The giveaway duration in days", false).setRequiredRange(1, MAX_TIMEOUT),
+					new OptionData(OptionType.INTEGER, WEEKS, "The giveaway duration in weeks", false).setRequiredRange(1, MAX_TIMEOUT),
+					new OptionData(OptionType.INTEGER, MONTHS, "The giveaway duration in months", false).setRequiredRange(1, MAX_TIMEOUT),
+					new OptionData(OptionType.INTEGER, YEARS, "The giveaway duration in years", false).setRequiredRange(1, MAX_TIMEOUT)),
+				new SubcommandData("end", "End a giveaway").addOption(OptionType.STRING, MESSAGE_ID, "The message id of the giveaway.", true)
+			))
+		);
+	}
+
+	private static final SimpleDateFormat SIMPLE_DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd_HH:mm");
```
Comment (RealYusufIsmail) [inline src/main/java/com/github/black0nion/blackonionbot/commands/fun/GiveawayCommand.java:57]: Put this at the top

**Review by RealYusufIsmail — CHANGES_REQUESTED** [review]

File: src/main/java/com/github/black0nion/blackonionbot/bot/SlashCommandBase.java [hunk 4]
```diff
@@ -34,28 +36,43 @@
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import javax.annotation.Nullable;
 import java.util.*;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Executors;
 import java.util.stream.Collectors;
 
 public class SlashCommandBase extends ListenerAdapter {
```
Comment (RealYusufIsmail) [inline src/main/java/com/github/black0nion/blackonionbot/bot/SlashCommandBase.java:45]: Add Java doc here

File: src/main/java/com/github/black0nion/blackonionbot/commands/SlashCommand.java [hunk 5]
```diff
@@ -22,8 +22,7 @@
 import static com.github.black0nion.blackonionbot.utils.Utils.gOD;
 
 @SuppressWarnings("unused")
-public abstract class SlashCommand extends GenericCommand {
-
+public abstract class SlashCommand {
```
Comment (RealYusufIsmail) [inline src/main/java/com/github/black0nion/blackonionbot/commands/SlashCommand.java:25]: Add Java doc here

File: src/main/java/com/github/black0nion/blackonionbot/commands/SlashCommandBuilder.java [hunk 6]
```diff
@@ -1,17 +1,20 @@
...
 
 import javax.annotation.Nonnull;
-import java.util.*;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
 
 @SuppressWarnings("unused")
 public class SlashCommandBuilder {
```
Comment (RealYusufIsmail) [inline src/main/java/com/github/black0nion/blackonionbot/commands/SlashCommandBuilder.java:20]: Add Java doc here

File: src/main/java/com/github/black0nion/blackonionbot/commands/SlashCommandEvent.java [hunk 7]
```diff
@@ -17,8 +17,10 @@
 import net.dv8tion.jda.api.entities.TextChannel;
 import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;
 import net.dv8tion.jda.api.interactions.InteractionHook;
+import org.slf4j.ILoggerFactory;
 import org.slf4j.LoggerFactory;
 
+import javax.annotation.Nonnull;
 import javax.annotation.Nullable;
 import java.util.function.Consumer;
 import java.util.stream.Collectors;
```
Comment (RealYusufIsmail) [inline src/main/java/com/github/black0nion/blackonionbot/commands/SlashCommandEvent.java:26]: Add Java doc

File: src/main/java/com/github/black0nion/blackonionbot/mongodb/MongoManager.java [hunk 8]
```diff
@@ -85,18 +85,4 @@ public static void updateValue(final MongoCollection<Document> collection, final
 			collection.insertOne(updatedValue);
 		}
 	}
-
```
Comment (RealYusufIsmail) [inline src/main/java/com/github/black0nion/blackonionbot/mongodb/MongoManager.java:88]: Add Java doc to this class.

**Review by RealYusufIsmail — CHANGES_REQUESTED** [review]
Other wise LGTM!

File: src/main/java/com/github/black0nion/blackonionbot/mongodb/MongoManager.java (hunk 8 above)
Comment (SIMULATAN) [reply src/main/java/com/github/black0nion/blackonionbot/mongodb/MongoManager.java:88]: I'm thinking about switching to PostgreSQL anyway, so I'll wait on this.

Comment (RealYusufIsmail) [reply src/main/java/com/github/black0nion/blackonionbot/mongodb/MongoManager.java:88]: ok

**Review by RealYusufIsmail — APPROVED** [review]
All good. You can merge the pr now.

**Pull request merged**